Source/player.cpp

```diff
@@ -538,31 +538,29 @@ void StartSpell(int pnum, Direction d, int cx, int cy)
 	player.spellLevel = GetSpellLevel(pnum, player._pSpell);
 }
 
-void RespawnDeadItem(Item *itm, Point target)
+void RespawnDeadItem(const Item &itm, Point target)
 {
 	if (ActiveItemCount >= MAXITEMS)
 		return;
 
 	int ii = AllocateItem();
+	Items[ii] = itm;
 
-	dItem[target.x][target.y] = ii + 1;
-
-	Items[ii] = *itm;
 	Items[ii].position = target;
-	RespawnItem(&Items[ii], true);
+	dItem[target.x][target.y] = ii + 1;
 
-	itm->_itype = ItemType::None;
+	RespawnItem(Items[ii], true);
 }
 
-void DeadItem(Player &player, Item *itm, Displacement direction)
+void DeadItem(Player &player, Item &itm, Displacement direction)
 {
-	if (itm->isEmpty())
+	if (itm.isEmpty())
 		return;
 
 	Point target = player.position.tile + direction;
 	if (direction != Displacement { 0, 0 } && ItemSpaceOk(target)) {
 		RespawnDeadItem(itm, target);
-		player.HoldItem = *itm;
+		player.HoldItem._itype = ItemType::None;

```

Comment:
```suggestion
		itm._itype = ItemType::None;
```

Reply:
I ended up rolling this back into RespawnDeadItem, it's meant to indicate that the item has been moved but because it's trivially copyable clang-tidy discourages actually using std::move :D

Reply:
And back to the caller 🙃 . Tried to tidy things up so it doesn't always assume that the item being dropped needs to be invalidated, many of the call sites are temporaries anyway.